### DISCUSSION TITLE
can't get service list

### DISCUSSION THREAD

**Discussion by ryudwig** [jaegertracing/jaeger-operator#1486] [Q&A]
hello, I installed Jaeger Operator v1.22.1 for Maistra Service Mesh.  (in OKD 4.6)
but I couldn't get service list in Jaeger UI.
I found log in jaeger-operator (in openshift-operators namespace)
"The service account running this operator does not have the role 'system:auth-delegator', consider granting it for additional capabilities"

Note) Recently, the kiali operator has been updated and is now operating normally.  https://github.com/kiali/kiali/discussions/4099

![image](https://user-images.githubusercontent.com/52852208/123235056-7db5df00-d516-11eb-9943-8117340fb91b.png)

![image](https://user-images.githubusercontent.com/52852208/123234932-5f4fe380-d516-11eb-8a5f-596580ec5b9e.png)

```
time="2021-06-22T15:48:56Z" level=info msg=Versions arch=amd64 identity=openshift-operators.jaeger-operator jaeger=1.22.0 jaeger-operator=v1.22.1 operator-sdk=v0.18.2 os=linux version=go1.14.15
I0622 15:48:57.397209       1 request.go:621] Throttling request took 1.037011928s, request: GET:https://172.30.0.1:443/apis/authentication.k8s.io/v1?timeout=32s
time="2021-06-22T15:49:02Z" level=info msg="Auto-detected the platform" platform=openshift
time="2021-06-22T15:49:02Z" level=info msg="Auto-detected ingress api" ingress-api=networking
time="2021-06-22T15:49:02Z" level=info msg="Automatically adjusted the 'es-provision' flag" es-provision=no
time="2021-06-22T15:49:02Z" level=info msg="Automatically adjusted the 'kafka-provision' flag" kafka-provision=no
time="2021-06-22T15:49:02Z" level=info msg=""
I0622 15:49:07.435341       1 request.go:621] Throttling request took 1.547014832s, request: GET:https://172.30.0.1:443/apis/authorization.openshift.io/v1?timeout=32s
time="2021-06-22T15:49:12Z" level=info msg="OAuthProxy ImageStream namespace and/or name not defined" name= namespace=
time="2021-06-23T01:08:26Z" level=info msg="Storage type not provided. Falling back to 'memory'" instance=jaeger namespace=istio-system
time="2021-06-23T01:38:09Z" level=info msg="Storage type not provided. Falling back to 'memory'" instance=jaeger namespace=istio-system
time="2021-06-23T01:38:24Z" level=error msg="failed to apply the changes" error="Operation cannot be fulfilled on routes.route.openshift.io \"jaeger\": the object has been modified; please apply your changes to the latest version and try again" execution="2021-06-23 01:38:23.789605347 +0000 UTC" instance=jaeger namespace=istio-system
```

**Comment by jpkrohling** [comment]
Which services do you expect to see there? How is your service mesh configured?

**Reply by ryudwig** [reply]
Thanks for your attention to my question.

I installed all operator in openshift-operators project include jaeger operator.
and I setuped istio control plane in istio-system project.
I would like to see all project that annotated to enable automatic injection of the Istio sidecar for Maistra Service Mesh ( like bookinfo, istio sample project at same cluster)

I have installed it several times in the past to check the normal operation of jaeger. It doesn't seem to work when installing with a combination of new versions. 
Currently only Istio and kiali are working fine. ^^;
Of course, the bookinfo application is receiving request/response through istio.

![image](https://user-images.githubusercontent.com/52852208/123292075-aad1b400-d54d-11eb-9c7c-9e8dc268efc9.png)

The installation and setting method was performed according to the link below.
https://maistra.io/docs/installing-ossm.html
https://maistra.io/docs/prepare-to-deploy-applications-ossm.html   (Bookinfo example application Section)

The important settings are as follows, and according to the guide, I did not set up jaeger directly.

1. add Project in ServiceMeshMemberRoll
```
apiVersion: maistra.io/v1
  kind: ServiceMeshMemberRoll
  metadata:
    name: default
  spec:
    members:
    - bookinfo
```
    
2. add annotation in Deployment
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: details-v1
  labels:
    app: details
    version: v1
spec:
  replicas: 1
  selector:
    matchLabels:
      app: details
      version: v1
  template:
    metadata:
      annotations:
        sidecar.istio.io/inject: "true"
        ```

**Reply by ryudwig** [reply]
is there anything wrong with my configuration?

**Reply by jpkrohling** [reply]
If you are not seeing traces from any services, I would double check your Maistra/Istio configuration.

**Reply by jkandasa** [reply]
@ryudwig @jpkrohling I could not reproduce this issue, If you provide the Maistra CR and detailed steps followed, might help us to find the root cause.

The steps I followed,
* Installed all the required operators via GUI
```bash
$ oc version
Client Version: openshift-clients-4.6.0-202006250705.p0-164-gffd683609
Server Version: 4.7.18
Kubernetes Version: v1.20.0+87cc9a4

$ oc get csv
NAME                      DISPLAY                     VERSION   REPLACES                  PHASE
jaeger-operator.v1.21.3   Community Jaeger Operator   1.21.3    jaeger-operator.v1.20.0   Succeeded
kiali-operator.v1.36.0    Kiali Operator              1.36.0    kiali-operator.v1.35.0    Succeeded
maistraoperator.v2.0.2    Maistra Service Mesh        2.0.2-0   maistraoperator.v2.0.1    Succeeded
```

```bash
$ oc new-project istio-system

$ oc create -f  smcp_2.0_all-in-one.yaml 
servicemeshcontrolplane.maistra.io/basic created

$ cat smcp_2.0_all-in-one.yaml 
apiVersion: maistra.io/v2
kind: ServiceMeshControlPlane
metadata:
  namespace: istio-system
  name: basic
spec:
  tracing:
    sampling: 10000
    type: Jaeger
  policy:
    type: Istiod
  addons:
    grafana:
      enabled: true
    jaeger:
      install:
        storage:
          type: Memory
    kiali:
      enabled: true
    prometheus:
      enabled: true
  version: v2.0
  telemetry:
    type: Istiod


$ oc new-project bookinfo

$ oc create -f member_roll.yaml
servicemeshmemberroll.maistra.io/default created

$ cat member_roll.yaml
apiVersion: maistra.io/v1
kind: ServiceMeshMemberRoll
metadata:
  name: default
  namespace: istio-system
spec:
  members:
    - bookinfo

$ oc get pods -n istio-system
NAME                                   READY   STATUS    RESTARTS   AGE
grafana-64d69c6574-p6bvn               2/2     Running   0          18m
istio-egressgateway-6bb7ff8897-t7wjs   1/1     Running   0          18m
istio-ingressgateway-cd5455b8d-v5v8s   1/1     Running   0          18m
istiod-basic-67b8b646dd-ckmvj          1/1     Running   0          19m
jaeger-7544cf4c5f-lwq6v                2/2     Running   0          18m
kiali-f79cb994d-9xkk7                  1/1     Running   0          44s
prometheus-98bfd6b5f-sx8vb             3/3     Running   0          18m

$ oc apply -n bookinfo -f https://raw.githubusercontent.com/Maistra/istio/maistra-2.0/samples/bookinfo/platform/kube/bookinfo.yaml
service/details created
serviceaccount/bookinfo-details created
deployment.apps/details-v1 created
service/ratings created
serviceaccount/bookinfo-ratings created
deployment.apps/ratings-v1 created
service/reviews created
serviceaccount/bookinfo-reviews created
deployment.apps/reviews-v1 created
deployment.apps/reviews-v2 created
deployment.apps/reviews-v3 created
service/productpage created
serviceaccount/bookinfo-productpage created
deployment.apps/productpage-v1 created

$ oc apply -n bookinfo -f https://raw.githubusercontent.com/Maistra/istio/maistra-2.0/samples/bookinfo/networking/bookinfo-gateway.yaml
gateway.networking.istio.io/bookinfo-gateway created
virtualservice.networking.istio.io/bookinfo created

$ export GATEWAY_URL=$(oc -n istio-system get route istio-ingressgateway -o jsonpath='{.spec.host}')

$ oc apply -n bookinfo -f https://raw.githubusercontent.com/Maistra/istio/maistra-2.0/samples/bookinfo/networking/destination-rule-all.yaml
destinationrule.networking.istio.io/productpage created
destinationrule.networking.istio.io/reviews created
destinationrule.networking.istio.io/ratings created
destinationrule.networking.istio.io/details created

$ oc get pods -n bookinfo
NAME                             READY   STATUS    RESTARTS   AGE
details-v1-86dfdc4b95-kp5rb      2/2     Running   0          98s
productpage-v1-658849bb5-cqtrs   2/2     Running   0          87s
ratings-v1-76b8c9cbf9-zm2rg      2/2     Running   0          95s
reviews-v1-58b8568645-kwb6s      2/2     Running   0          92s
reviews-v2-5d8f8b6775-fjf4v      2/2     Running   0          91s
reviews-v3-666b89cfdf-qvktt      2/2     Running   0          90s

$ curl -o /dev/null -s -w "%{http_code}\n" http://$GATEWAY_URL/productpage
200
```

Screenshots:
* Kiali UI
![kiali-tracing](https://user-images.githubusercontent.com/1004403/124127903-e31b4a00-da99-11eb-9550-33b672d14bde.png)

* Jaeger UI
![jaeger_tracing](https://user-images.githubusercontent.com/1004403/124127886-e0b8f000-da99-11eb-96e2-3aa35fb78a1b.png)

**Reply by ryudwig** [reply]
thank you. I will retry

**Reply by ryudwig** [reply]
@jkandasa 
I did it the way you told me. (operator was not reinstalled)
All service is still not visible in jaeger.
The only difference between you and me seems to be the version of Jaeger operation.
jaeger-operator.v1.21.3 (yours) vs. jaeger-operator.v1.22.1 

$ oc get csv
NAME                       DISPLAY                        VERSION   REPLACES                   PHASE
elastic-cloud-eck.v1.6.0   Elasticsearch (ECK) Operator   1.6.0     elastic-cloud-eck.v1.5.0   Succeeded
jaeger-operator.v1.22.1    Community Jaeger Operator      1.22.1    jaeger-operator.v1.21.3    Succeeded
kiali-operator.v1.36.0     Kiali Operator                 1.36.0    kiali-operator.v1.35.0     Succeeded
maistraoperator.v2.0.2     Maistra Service Mesh           2.0.2-0   maistraoperator.v2.0.1     Succeeded

I will retry from installing all operators.
